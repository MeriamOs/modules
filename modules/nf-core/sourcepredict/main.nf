process SOURCEPREDICT {
    tag "$meta.id"
    label 'process_medium'

    conda "${moduleDir}/environment.yml"
    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
        'https://depot.galaxyproject.org/singularity/sourcepredict:0.5.1--pyhdfd78af_0':
        'biocontainers/sourcepredict:0.5.1--pyhdfd78af_0' }"

    input:
    tuple val(meta), path(kraken_parse)
    path sources
    path labels

    output:
    tuple val(meta), path("*.sourcepredict.csv"), emit: report
    path "versions.yml"           , emit: versions

    when:
    task.ext.when == null || task.ext.when

    script:
    def args = task.ext.args ?: ''
    def prefix = task.ext.prefix ?: "${meta.id}"

    """
    sourcepredict \\
        -s $sources \\
        -l $labels \\
        $args \\
        -t $task.cpus \\
        -o ${prefix}.sourcepredict.csv \\
        ${kraken_parse}

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        sourcepredict: \$(echo \$(sourcepredict --help 2>&1 | head -n 1 | sed -n "s/.*SourcePredict v\\([0-9.]*\\).*/\\1/p")
    END_VERSIONS
    """

    stub:
    def args = task.ext.args ?: ''
    def prefix = task.ext.prefix ?: "${meta.id}"

    """
    touch ${prefix}.sourcepredict.csv

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        sourcepredict: \$(echo \$(sourcepredict --help 2>&1 | head -n 1 | sed -n "s/.*SourcePredict v\\([0-9.]*\\).*/\\1/p")
    END_VERSIONS
    """
}
