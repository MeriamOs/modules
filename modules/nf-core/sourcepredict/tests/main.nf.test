nextflow_process {

    name "Test Process SOURCEPREDICT"
    script "../main.nf"
    process "SOURCEPREDICT"

    tag "modules"
    tag "modules_nfcore"
    tag "sourcepredict"

    test("metagenome - csv") {

        when {
            process {
                """                
                input[0] = [
                    [ id:'test' ], // meta map
                    file('/nesi/nobackup/uoo02328/meriam/sourcepredict/dog_exsample.csv', checkIfExists: true),
                ]
                input[1] = [
                    [ id:'sources' ],
                    file('/nesi/nobackup/uoo02328/meriam/sourcepredict/sp_sources.csv', checkIfExists: true)
                ]
                input[2] = [
                    [ id:'labels' ],
                    file('/nesi/nobackup/uoo02328/meriam/sourcepredict/sp_labels.csv', checkIfExists: true)
                ]
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(
                    process.out.report,
                    process.out.versions,
                ).match()
                }
            )
        }

    }

    test("metagenome - csv - stub") {

        options "-stub"

        when {
            process {
                """                
                input[0] = [
                    [ id:'test' ], // meta map
                    file('/nesi/nobackup/uoo02328/meriam/sourcepredict/dog_exsample.csv', checkIfExists: true),
                ]
                input[1] = [
                    [ id:'sources' ],
                    file('/nesi/nobackup/uoo02328/meriam/sourcepredict/sp_sources.csv', checkIfExists: true)
                ]
                input[2] = [
                    [ id:'labels' ],
                    file('/nesi/nobackup/uoo02328/meriam/sourcepredict/sp_labels.csv', checkIfExists: true)
                ]
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(
                    process.out.report,
                    process.out.versions,
                ).match()
                }
            )
        }

    }

}
