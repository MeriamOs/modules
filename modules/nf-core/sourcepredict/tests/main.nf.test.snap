{
    "metagenome - csv - stub": {
        "content": [
            [
                [
                    {
                        "id": "test"
                    },
                    "test.sourcepredict.csv:md5,d41d8cd98f00b204e9800998ecf8427e"
                ]
            ],
            [
                "versions.yml:md5,d0e1400b2ad38aa0f76376e617a80015"
            ]
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "24.04.4"
        },
        "timestamp": "2024-10-23T14:30:12.413637144"
    },
    "metagenome - csv": {
        "content": [
            [
                [
                    {
                        "id": "test"
                    },
                    "test.sourcepredict.csv:md5,fd90e16e7d7ce577d9b561afb1ebb24b"
                ]
            ],
            [
                "versions.yml:md5,d0e1400b2ad38aa0f76376e617a80015"
            ]
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "24.04.4"
        },
        "timestamp": "2024-10-23T14:29:42.591511161"
    }
}